pkg/infrastructure/gcp/clusterapi/network.go

```diff
@@ -85,9 +109,74 @@ func createInternalLBAddress(ctx context.Context, in clusterapi.InfraReadyInput)
 		},
 	}
 
-	if _, err := service.HealthChecks.Insert(in.InstallConfig.Config.GCP.ProjectID, healthCheck).Context(ctx).Do(); err != nil {
+	_, err = service.RegionHealthChecks.Insert(projectID, region, healthCheck).Context(ctx).Do()
+	if err != nil {
```

Comment:
I'm wondering if we need to augment the delete code, I see this getting created but when I destroy the cluster I don't see it being deleted. Only this health check resource remains e.g.:
```
$ gcloud compute health-checks describe https://www.googleapis.com/compute/v1/projects/openshift-dev-installer/regions/us-east1/healthChecks/bfournie-capg-test-gn6g7-api-internal
checkIntervalSec: 2
creationTimestamp: '2024-04-06T10:28:41.909-07:00'
description: Created By OpenShift Installer
healthyThreshold: 3
httpsHealthCheck:
  port: 6443
  proxyHeader: NONE
  requestPath: /readyz
id: '7659072829660084390'
kind: compute#healthCheck
name: bfournie-capg-test-gn6g7-api-internal
region: https://www.googleapis.com/compute/v1/projects/openshift-dev-installer/regions/us-east1
selfLink: https://www.googleapis.com/compute/v1/projects/openshift-dev-installer/regions/us-east1/healthChecks/bfournie-capg-test-gn6g7-api-internal
timeoutSec: 2
type: HTTPS
unhealthyThreshold: 3
```

Reply:
Added 7de950d175b37042926d8429d811753b5d44c7fa to delete HTTPS health checks. Before we were only handling HTTP health checks. Have not tested yet, but will now.

Reply:
> Added [7de950d](https://github.com/openshift/installer/commit/7de950d175b37042926d8429d811753b5d44c7fa) to delete HTTPS health checks. Before we were only handling HTTP health checks. Have not tested yet, but will now.

Hm no, tested and I don't think this is the right way. The CAPG-created https is getting deleted, but the internal one is still not. will debug that soon.

Also my install failed with:

```
ERROR failed to fetch Cluster: failed to generate asset "Cluster": failed to create cluster: failed preparing ignition data: ignition failed to provision storage: failed to create storage: failed to create bucket: googleapi: Error 409: Your previous request to create the named bucket succeeded and you already own it., conflict 
```

I suspect #8056 broke the capg flow. #8056 adds ignition bucket creation in tfvars, which is still executed in the capi flow, so I think the same logic is being run twice and we hit this error. cc @barbacbd

Reply:
Left further comments on how to fix https://github.com/openshift/installer/pull/8056#issuecomment-2041659263